generators/base-core/generator.ts

```diff
@@ -257,6 +257,12 @@ export default class CoreGenerator extends YeomanGenerator<JHipsterGeneratorOpti
     return this._needleApi;
   }
 
+  get control(): Control {
```

Comment:
Not a fan of methods with stateful objects in generator itself. It simplifies the calls, but I would prefer that the element being injected as a parameter of the tasks that needs it

Reply:
Yes it will be only possible to use in tasks parameters.
This was required by old needles api implementation.

https://github.com/jhipster/generator-jhipster/pull/29501 converts it to private so it won’t be available.